### DISCUSSION TITLE
Encoding question

### DISCUSSION THREAD

**Discussion by JenyaRostov** [zyantific/zydis#363] [Q&A] [answered]
Less of an issue and more of a question
How do i encode this:
mov eax, [someAddress]
i tried modifying `ZydisEncoderRequest` with this instruction, but changed displacement to a much larger value
from 503640 to 137616392292062
what do i need to change in my request? (request is generated from `ZydisDecodedInstruction`)

**Comment by athre0z** [comment]
Please provide your code and let us know how you determined the "incorrect" displacement to allow us to get a better idea of what is going wrong here. I suspect that this is likely due to `rip` relative addressing, but can't say for sure without further information.

**Comment by JenyaRostov** [comment, marked as answer]
kinda figured it out
my original instruction was with a small displacement and encoded as follows: 8B 05 4E650800 mov eax, [variable]
zydis correctly identified it as `rip` displacement
but to change displacement to a much larger value i had to this
![image](https://user-images.githubusercontent.com/52300935/178297016-a2958a78-b262-4d1d-91d7-28acc59be54a.png)
which changes instruction to A1 E886D391F77F0000  (absolute address)

**Reply by JenyaRostov** [reply]
oops should've written it as reply

**Reply by mappzor** [reply]
If you want to remain RIP-rel you should set `ZYDIS_REGISTER_RIP` as `mem.base`. Best solution is to use `ZydisEncoderDecodedInstructionToEncoderRequest` and modify `requests.operands[1].mem.displacement`.

> i tried modifying ZydisEncoderRequest with this instruction, but changed displacement to a much larger value
from 503640 to 137616392292062

RIP-rel displacements are limited to 32 bits and it's signed, so you cannot reference anything beyond -2GB and +2GB from RIP.
